### PR TITLE
Android T patches

### PR DESCRIPTION
**Pull request by SaliniVenate**
Port the bsp_diff patches introduced in Android-T to corresponding github projects

### CODE REVIEW

File: _studio/shared/umc/core/umc/include/umc_structures.h [hunk 1]
```diff
@@ -691,7 +691,7 @@ namespace UMC
     {
         // some compilers complain to conversion to/from
         // pointer types from/to integral types.
-        return (T) ((((uint8_t *) pv - (uint8_t *) 0) + (lAlignValue - 1)) &
```
Comment (TianmiChen) [inline _studio/shared/umc/core/umc/include/umc_structures.h:694]: Why do we have changes here? Please explain.

Comment (SaliniVenate) [reply _studio/shared/umc/core/umc/include/umc_structures.h:694]: This change was introduced to fix below build error:

Clang-14 complains about an unusual way of converting
a pointer to an integer:

error: invalid operands to binary expression
('uintptr_t *' (aka 'unsigned long *') and 'unsigned long')
   return (T) ((((uintptr_t *) pv) + (lAlignValue - 1)) &

Replace this with a normal cast to uintptr_t

Comment (SaliniVenate) [reply _studio/shared/umc/core/umc/include/umc_structures.h:694]: I created the patch based on this patch: https://www.spinics.net/lists/stable-commits/msg218632.html

Comment (TianmiChen) [reply _studio/shared/umc/core/umc/include/umc_structures.h:694]: okay, thanks for the explanation.

**Review by TianmiChen — APPROVED** [review]

**Review by dyang23 — APPROVED** [review]
LGTM

**Pull request merged**